src/ovs_events.c

```diff
@@ -571,82 +693,118 @@ static void ovs_events_conn_initialize(ovs_db_t *pdb) {
       return;
     }
   }
-  OVS_EVENTS_CTX_LOCK { ovs_events_ctx.is_db_available = 1; }
-  DEBUG(OVS_EVENTS_PLUGIN ": OVS DB connection has been initialized");
+  ovs_events_inst_t *inst = user_data;
+  OVS_EVENTS_LOCK(&inst->mutex) { inst->ovs_db = pdb; }
+  DEBUG(OVS_EVENTS_PLUGIN ": %s: OVS DB connection has been initialized",
+        inst->name);
 }
 
 /* OVS DB terminate connection notification callback */
-static void ovs_events_conn_terminate() {
-  const char msg[] = "OVS DB connection has been lost";
-  if (ovs_events_ctx.config.send_notification)
-    ovs_events_dispatch_terminate_notification(msg);
-  WARNING(OVS_EVENTS_PLUGIN ": %s", msg);
-  OVS_EVENTS_CTX_LOCK { ovs_events_ctx.is_db_available = 0; }
+static void ovs_events_conn_terminate(ovs_db_t *pdb, void *user_data) {
+  char *msg = "OVS DB connection has been lost";
+  ovs_events_inst_t *inst = user_data;
+  if (ovs_events_ctx.send_notification)
+    ovs_events_dispatch_terminate_notification(inst->name, msg);
+  WARNING(OVS_EVENTS_PLUGIN ": %s: %s", inst->name, msg);
+  OVS_EVENTS_LOCK(&inst->mutex) { inst->ovs_db = NULL; }
 }
 
 /* Read OVS DB interface link status callback */
-static int ovs_events_plugin_read(__attribute__((unused)) user_data_t *u) {
-  _Bool is_connected = 0;
-  OVS_EVENTS_CTX_LOCK { is_connected = ovs_events_ctx.is_db_available; }
-  if (is_connected)
-    if (ovs_db_send_request(ovs_events_ctx.ovs_db, "transact",
-                            ovs_events_ctx.ovs_db_select_params,
-                            ovs_events_poll_result_cb) < 0) {
-      ERROR(OVS_EVENTS_PLUGIN ": get interface info failed");
-      return -1;
+static int ovs_events_plugin_read(user_data_t *u) {
+  int status = 0;
+  ovs_events_inst_t *inst = u->data;
+  OVS_EVENTS_LOCK(&inst->mutex) {
+    if (inst->ovs_db) {
+      if (ovs_db_send_request(inst->ovs_db, "transact",
+                              inst->config.select_params,
+                              ovs_events_poll_result_cb) < 0) {
+        ERROR(OVS_EVENTS_PLUGIN ": get interface info failed");
+        status = (-1);
```

Comment:
Please either remove the braces, or remove the status variable all together and directly return -1.